tests/test_visitors/test_ast/test_classes/test_getter_setter.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.oop import (
+    UnpythonicGetterSetterViolation,
+)
+from wemake_python_styleguide.visitors.ast.classes import WrongClassVisitor
+
+module_getter_and_setter = """
+attribute = 1
+
+def get_attribute():
+    ...
+
+def set_attribute():
+    ...
+"""
+
+static_getter_and_setter = """
+attribute = 1
+
+class Test(object):
+    @staticmethod
+    def get_attribute():
+        ...
+
+    @staticmethod
+    def set_attribute():
+        ...
+"""
+
+property_getter_and_setter = """
+class Test(object):
+    def __init__(self):
+        self.attribute = 1
+
+    @property
+    def get_attribute(self):
+        ...
+
+    @property.setter
+    def set_attribute(self):
+        ...
+"""
+
+child_getter_and_setter = """
+class TestParent(object):
+    def __init__(self):
+        self.attribute = 1
+
+class TestChild(TestParent):
+    def get_attribute(self):
+        ...
+
+    def set_attribute(self):
+        ...
+"""
+
+nested_getter_and_setter = """
+class Template(object):
+    def __init__(self):
+        self.attribute = 1
+
+    def some_function(self):
+        def get_attribute(self):
+            ...
+        def set_attribute(self):
+            ...
+        get_attribute(self)
+"""
+
+class_attribute_template = """
+class Template(object):
+    def __init__(self):
+        self.{0} = 1
```

Comment:
We also need to cover class attributes here, in forms of:

- AnnAssign like `attribute: int`
- AnnAssign like `attribute: int = 1`
- Assign like `attribute = 1`
- Assign like `attribute = other = 1`
- Assign like `attribute, other = 1, 2`

Instance cases not covered:

- AnnAssign like `self.attribute: int = 1`
- Assign like `self.attribute = self.other = 1`
- Assign like `self.attribute, self.other = 1, 2`

We also need to test every case with public, protected, and private access modificators.